src/api_client/api_client.js

```diff
@@ -84,11 +85,17 @@ export default class APIClient {
 			throw new Error('APIClient requires nodes for initialization.');
 		}
 
+		if (typeof providedOptions !== 'object' || Array.isArray(providedOptions)) {
+			throw new Error(
+				'APIClient requires the second parameter to be an object.',
```

Comment:
We should make clear that it's optional. E.g. `APIClient takes an optional object as the second parameter.`?